app/models/button_maker.rb

```diff
@@ -17,19 +17,12 @@
 # pulling it all into a special-purpose, encapsulated place.
 # You probably don't want to, but when all else fails familiarize yourself with:
 # https://mitlibraries.atlassian.net/wiki/spaces/DI/pages/58654721/ButtonMaker+documentation
-class ButtonMaker
+module ButtonMaker
```

Comment:
Now that we've factored out all kinds of stuff (🎉 ), there should be some kind of comment here as to the types of interfaces child classes are expected to expose - in particular, that they must all define `html_button`, which returns either `nil` or HTML suitable for enabling this function, and they must all determine whether to return the button or `nil` (using a function `eligible?` unless there's some really good reason for it to be more complicated than that). And if there's anything interesting about why functionality is here rather than in child classes, that should also be addressed (although on first glance maybe the separation of functions is pretty logical and doesn't require explanation).